scripts/cba/clean_projects.py

```diff
@@ -0,0 +1,131 @@
+
+**Outputs**
+
+- ``resources/cba/transmission_projects.csv``: Cleaned CSV with columns:
+  - ``project_id``: Integer project identifier
+  - ``project_name``: Project name
+  - ``in_reference2030``: Boolean indicating if project is in 2030 reference grid
+  - ``in_reference2035``: Boolean indicating if project is in 2035 reference grid
+  - ``border``: Border string in format "BUS0-BUS1"
+  - ``p_nom 0->1``: Transfer capacity increase from bus0 to bus1 (MW)
+  - ``p_nom 1->0``: Transfer capacity increase from bus1 to bus0 (MW)
+  - ``bus0``: Source bus code (4 alphanumeric characters)
+  - ``bus1``: Destination bus code (4 alphanumeric characters)
+
+- ``resources/cba/storage_projects.csv``: Empty CSV with columns project_id and project_name (stub implementation)
+
+"""
+
+import logging
+from pathlib import Path
+
+import pandas as pd
+
+from scripts._helpers import configure_logging, set_scenario_config
+
+logger = logging.getLogger(__name__)
+
+TRANSMISSION_PROJECTS_COLUMN_MAP = {
+    "Project ID": "project_id",
+    "Project Name": "project_name",
+    "Is the project in the 2030 reference grid?": "in_reference2030",
+    "Is the project in the 2035 reference grid?": "in_reference2035",
+    "Border": "border",
+    "Transfer capacity increase A-B (MW)": "p_nom 0->1",
+    "Transfer capacity increase B-A (MW)": "p_nom 1->0",
+}
+
+
+def extract_transmission_projects(excel_path: Path) -> pd.DataFrame:
+    projects = (
+        pd.read_excel(
+            excel_path,
+            sheet_name="Trans.Projects",
+            skiprows=1,
+            usecols=list(TRANSMISSION_PROJECTS_COLUMN_MAP),
+            na_values=["  "],
+            true_values=["True", "WAHR"],
+            false_values=["False", "FALSCH"],
+        )
+        .rename(columns=TRANSMISSION_PROJECTS_COLUMN_MAP)
+        .dropna(how="all")  # contains many all nan rows
+    )
+    projects["project_id"] = projects["project_id"].astype(int)
+
+    # The multiple lines columns in the "Expected transfer capacity increase" section
+    # are exploded into separate rows
+    multiline_columns = ["border", "p_nom 0->1", "p_nom 1->0"]
+    projects = projects.assign(
+        **{c: projects[c].str.split("\n") for c in multiline_columns}
+    ).explode(multiline_columns)
+
+    projects = projects.assign(
+        **projects["border"].str.extract(
+            r"(?P<bus0>[A-Z0-9]{4})-(?P<bus1>[A-Z0-9]{4})$"
+        )
+    )
+
+    unclear_border = projects["bus0"].isna() | projects["bus1"].isna()
+    logger.warning(
+        "%d out of %d projects do not follow the simple <bus0>-<bus1> format, ignoring them:\n%s",
+        unclear_border.sum(),
+        len(unclear_border),
+        projects.loc[
+            unclear_border, ["project_id", "project_name", "border"]
+        ].to_string(index=False, max_colwidth=40, line_width=100),
+    )
+    projects = projects.loc[~unclear_border]
+    return projects
```

Comment:
There are a few project without any given capacity increase in any direction, e.g. `HansaLink - Phase 1 `. There should  be one additional check to filter these projects out with a warning

Reply:
done